### DISCUSSION TITLE
Can you please tell me the name of API for creating pod and setting quota for that pod, in Kubeapps API documentation?

### DISCUSSION THREAD

**Discussion by sehrishnaseer** [kubeapps/kubeapps#3930] [Q&A] [answered]
Can you please tell me the name of API for creating pod and setting quota for that pod, in Kubeapps API documentation? And will the .yaml file will be automatically created once the pod is created?

**Comment by antgamdia** [comment, marked as answer]
Hi @sehrishnaseer, I'm afraid I'm not fully getting your question. The latest API documentation is written in this [OpenAPI doc file](https://raw.githubusercontent.com/kubeapps/kubeapps/master/dashboard/public/openapi.yaml); however, any of the endpoints will create a pod or set a quota.
In Kubeapps you can manage _packaged_ applications (like Helm, Flux, Carvel or Operators). The latest packages API we are building is made of endpoints like: _GetAvailablePackageSummaries_ or _CreateInstalledPackage_, but we don't manage pods directly. 
Please feel free to add more information so that I can point you out to a better solution.

**Reply by sehrishnaseer** [reply]
Can you please tell me in detail, if I can get your precious time, that which API is performing which function? And specifically which end points are you talking about to create pod and set quota? And is there any API to create deployment too?

**Reply by antgamdia** [reply]
Sure, let me quickly walk you through our main APIs. Also, note that every method can be invoked directly from a Kubeapps installation with the interactive API documentation we provide at the bottom right side menu:

![image](https://user-images.githubusercontent.com/11535726/145954572-6333e968-0d19-4427-b579-d30856139710.png)

Disclaimer: we are in an ongoing process of revamping our backend to support different plugins (Helm, Carvel, Flux...) more easily, therefore, some endpoints may disappear in future versions (`kubeops` mainly). The Kubeapps API is not considered stable yet.

Here's an overview of the different APIs we have:

![image](https://user-images.githubusercontent.com/11535726/145955074-bd55bf1b-af6c-4277-8e87-ac3ecb6bd16f.png)

### Deprecated endpoints:

- **Kubeops**: was the component in charge of interacting with Helm releases and some Kuebernetes API resources (like namespaces, or can-i endpoints).
- "**Kubernetes API**" is the actual Kubernetes API server being proxied, so that an authenticated user in Kubeapps will be able to perform authenticated calls to the Kubernetes API. We are actively deprecating this proxied API, but as of today there are some remaining aspects to tackle (operators, for instance).

### New API

- **Plugin service**: an API for querying the plugins that are running in a particular Kubeapps installation. 
- **Packages service**: is a package-agnostic API for performing the main primitives we have defined:
    - **Available packages** (packages you can potentially install)
        - GET /availablepackages: retrieve the summary of the available packages for every enabled plugin
        - GET  /availablepackages/<plugin name, plugin version, cluster, namespace...>/<packageID>: retrieves the details of a particular available package
        - GET  /availablepackages/<plugin name, plugin version, cluster, namespace...>/<packageID>/versions: retrieves all the possible versions for an available package
    - **Installed packages** (packages you already installed)
        - GET /installedpackages: retrieve the summary of the installed packages for every enabled plugin
        - GET /installedpackages/<plugin name, plugin version, cluster, namespace...>/<packageID>: retrieves the details of a particular installed package
        - POST /installedpackages: install a new package
        - DELETE /installedpackages/<plugin name, plugin version, cluster, namespace...>/<packageID>: deletes a particular installed pakage
        - PUT /installedpackages/<plugin name, plugin version, cluster, namespace...>/<packageID>: updates a particular installed package


- **FluxV2Packages** service: implements exactly the same operations the Packages service does, but only interacts with packages  managed by Fluxv2 (that is, Fluxv2 Custom Resources)
- **HelmV2Packages** service: implements exactly the same operations the Packages service does, but only interacts with packages managed by Helm  (that is, Helm Charts and Helm Releases)
- **KappControllerPackages** service: implements exactly the same operations the Packages service does, but only interacts with packages managed by Kapp-controller  (that is, Carvel Custom Resources).

Please find more technical details and discussion in our [design document](https://docs.google.com/document/d/1z03msZRGsRvcRaom-yrvEwIWcEfNy6fSA5Zg28gjYvA/edit#heading=h.yheooaol1kzd) or in this [video explaining the new Kubeapps API](https://tanzu.vmware.com/developer/tv/tanzu-tuesdays/0074/).


Finally, regarding your specific question:

> And specifically which end points are you talking about to create pod and set quota? And is there any API to create deployment too?

We don't have any API endpoint for creating _standalone_ pods or deployments and setting their quotas. You need to package your application beforehand using [Helm](https://helm.sh/docs/helm/helm_create/) or [Carvel](https://carvel.dev/kapp-controller/).

Hope this answers your question :)
